config/ogdl/src/test/groovy/org/apache/shiro/config/ogdl/ReflectionBuilderTest.groovy

```diff
@@ -640,6 +643,20 @@ class ReflectionBuilderTest {
         assertNotNull(beanMap.get("two"))
     }
 
+    @Test
+    void testNotMultipleInitialization() {
+        // given
+        Map<String, String> defs = new ConcurrentHashMap<>()
+        defs.put("initcountbean", InitCountBean.getCanonicalName())
+        ReflectionBuilder builder = new ReflectionBuilder()
+
+        // when
+        builder.buildObjects(defs);
+
+        // then
+        assertEquals(1, InitCountBean.getInitCount())
+    }
+
```

Comment:
Can you add a test where `buildObjects(null)` is called FIRST? Because that actually happens in Shiro.